### DISCUSSION TITLE
Delete internet permissions and access network state

### DISCUSSION THREAD

**Discussion by qw123wh** [WrichikBasu/ShakeAlarmClock#37] [Feature requests]
Other open source clock apps work fine even without internet permissions and wifi state access, it would be nice to do the same with your app

**Comment by WrichikBasu** [comment]
The app itself does not require internet connectivity for its main function. I have implemented a [library](https://github.com/javiersantos/AppUpdater) so that users can be notified when updates are available. That library asks for the internet permissions.

Also, in the `README.md`, it is written that the app works offline. In fact, I use it daily on a phone that does not have any SIM cards and is not connected to WiFi.

**Comment by qw123wh** [comment]
Ok I understand that library is only used to notify an app update, right?  However, you can also put it on F-Droid as well as GitHub and play store, so those who install it from F-Droid will receive the update notification from F-Droid, without the need for this library.  Anyway thanks for the explanation.

**Reply by WrichikBasu** [reply]
> Ok I understand that library is only used to notify an app update, right?

Yes, only for updates.

> However, you can also put it on F-Droid as well as GitHub and play store, so those who install it from F-Droid will receive the update notification from F-Droid, without the need for this library.

I will put the app on F-Droid soon, and in that case, I will disable the library.

**Comment by qw123wh** [comment]
Thank you so much.
